src/users/usersRouter.js

```diff
@@ -16,11 +17,52 @@ router.get('/', function (req, res) {
 
 // Create new user
 router.post('/create', function (req, res) {
-    Users.createUser(req.body)
-        .then((response) => {
-            res.status(200).json(response);
+    const auth0ReqBody = {
+        client_id: `${process.env.AUTH0_CLIENT_ID}`,
+        client_secret: `${process.env.AUTH0_CLIENT_SECRET}`,
+        audience: `${process.env.AUTH0_ISSUER_URL}api/v2/`,
+        grant_type: "client_credentials"
+    }
+    // GET TOKEN FROM AUTH0
+    axios.post(`${process.env.AUTH0_ISSUER_URL}oauth/token`, auth0ReqBody)
+        .then(tokenRes => {
+            let headers = {
+                'Authorization': `Bearer ${tokenRes.data.access_token}`
+            }
+            let authReqBody = {
+                email: req.body.email,
+                name: req.body.name,
+                password: req.body.password,
+                email_verified: false,
+                connection: "Username-Password-Authentication"
+            }
+            // CREATE USER ON AUTH0
+            axios.post(`${process.env.AUTH0_ISSUER_URL}api/v2/users`, authReqBody, { headers: headers })
+                .then(authRes => {
+                    let newUser = {
+                        auth0_id: authRes.data.user_id,
```

Comment:
Wouldn't the id be empty? I believe the auth0_id generated by auth0 when the user is created, so will this be an empty value? Are we assigning this a value on our end?

Reply:
Ah, so line 40 is where we send the user data up to Auth0. When Auth0 creates the user, it assigns them an Auth0 ID and returns it in the positive response (`authRes`), allowing us to grab that ID well before actually adding the user's info to our database!

This was super cool to finally find, because this was a major hurdle we struggled to overcome on Underdog Devs, haha.